ddsreplayer/test/blackbox/mcap/McapFileReadTest.cpp

```diff
@@ -0,0 +1,199 @@
+                    "Error running subscriber. Error message:\n " <<
+                        e.what());
+                }
+                close_handler_subscriber->simulate_event_occurred();
+            });
+
+
+    // Configuration File path
+    std::string file_path = "resources/config_file.yaml";
+    eprosima::ddsrecorder::yaml::ReplayerConfiguration configuration(file_path);
+
+    // Create replayer instance
+    std::string input_file = "resources/helloworld_file.mcap";
+    auto replayer = std::make_unique<DdsReplayer>(configuration, input_file);
+
+    std::cout << "replayer created !!!!" << std::endl;
+
+    // Start replaying data
+    bool read_success;
+    std::thread process_mcap_thread([&]
+            {
+                try
+                {
+                    replayer->process_mcap();
+                    read_success = true;
+                }
+                catch (const eprosima::utils::InconsistencyException& e)
+                {
+                    logError(DDSREPLAYER_ERROR,
+                    "Error processing MCAP file. Error message:\n " <<
+                        e.what());
+                    read_success = false;
+                }
+                close_handler_replayer->simulate_event_occurred();
+            });
+
+    // Wait until signal arrives (or all messages in MCAP file sent)
+    close_handler_replayer->wait_for_event();
+
+    // Wait until signal arrives
+    close_handler_subscriber->wait_for_event();
+
+    // Disable inner pipe, which would abort replaying messages in case execution stopped by signal
+    replayer->stop();
+
+    run_subscriber.join();
+    process_mcap_thread.join();
+
+    std::cout << "threads joined!!!!" << std::endl;
+
+    std::cout << "process info..." << std::endl;
+}
+
+TEST(McapFileReadTest, trivial)
+{
+    // info to check
+    DataToCheck data;
+    create_subscriber_replayer(data);
+    ASSERT_TRUE(true);
+}
+
+TEST(McapFileReadTest, n_msgs)
+{
+    // info to check
+    DataToCheck data;
+    create_subscriber_replayer(data);
+    ASSERT_EQ(data.n_received_msgs, 11);
+}
+
+TEST(McapFileReadTest, msg_type)
+{
+    // info to check
+    DataToCheck data;
+    create_subscriber_replayer(data);
+    ASSERT_EQ(data.type_msg, "HelloWorld");
+}
+
+TEST(McapFileReadTest, msg_message)
+{
+    // info to check
+    DataToCheck data;
+    create_subscriber_replayer(data);
+    ASSERT_EQ(data.message_msg, "Hello World");
+}
+
+TEST(McapFileReadTest, msg_min_index)
+{
+    // info to check
+    DataToCheck data;
+    create_subscriber_replayer(data);
+    ASSERT_EQ(data.min_index_msg, 0);
+}
+
+TEST(McapFileReadTest, msg_max_index)
+{
+    // info to check
+    DataToCheck data;
+    create_subscriber_replayer(data);
+    ASSERT_EQ(data.max_index_msg, 10);
+}
+
+TEST(McapFileReadTest, msg_hz)
+{
+    // info to check
+    DataToCheck data;
+    create_subscriber_replayer(data);
+    // hz ~ 200
+    ASSERT_GT(data.hz_msgs, 180);
+    ASSERT_LT(data.hz_msgs, 220);
+}
```

Comment:
I have the feeling these tests all test the same thing, but each of them focus on a different part.
What I believe should be done is to have different tests testing different features of the replayer; playback rate, begin/end time, start replay time, qos preservation, replaying dynamic types, etc.

Reply:
I also noted that these tests required reliability qos, which will depend on what qos were recorded (don't know if that's the case for this particular mcap file), although can be enforced by using the builtin-topics list.